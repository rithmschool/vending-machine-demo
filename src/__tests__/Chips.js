import React from 'react'; // for JSX
import { mount } from 'enzyme'; // how to mount the component
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json'; // for snapshotting the component
import Chips from '../Chips'; // import the component itself

describe('<Chips />', () => {
  let wrapper;
  it('renders', () => {
    wrapper = mount(
      <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
        <Chips />
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('responds to click', () => {
    const nomButton = wrapper.find('#nomnom');
    nomButton.simulate('click');
    expect(wrapper.find('.bag')).toHaveLength(1);
    nomButton.simulate('click');
    expect(wrapper.find('.bag')).toHaveLength(2);
  });
});
