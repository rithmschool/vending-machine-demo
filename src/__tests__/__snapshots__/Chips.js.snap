// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chips /> matches snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Chips>
      <div
        className="Chips"
      >
        <Message>
          <div
            className="Message"
            id="messagey"
          >
            <h1>
              bags eaten: 
              0
            </h1>
            <button
              id="nomnom"
              onClick={[Function]}
            >
              nom nom nom
            </button>
            <h1>
              <Link
                replace={false}
                to="/"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  go back
                </a>
              </Link>
            </h1>
          </div>
        </Message>
      </div>
    </Chips>
  </Router>
</MemoryRouter>
`;
